tools/export/nb/__init__.py

```diff
@@ -0,0 +1,402 @@
+
+        profiles = {}
+
+        for fn in file_names:
+            content = load(open(fn))
+            profile_name = basename(fn).replace(".json", "")
+            profiles[profile_name] = content
+
+        return profiles
+
+    @staticmethod
+    def get_netbeans_file_list(file_list):
+        output = []
+        prev_dir = ''
+        folder_count = 1
+        for idx, item in enumerate(file_list):
+            cur_dir = os.path.dirname(item)
+            dir_temp = os.path.normpath(cur_dir)
+            prev_dir_temp = os.path.normpath(prev_dir)
+            dir_list = dir_temp.split(os.sep)
+            prev_dir_list = prev_dir_temp.split(os.sep)
+            dir_depth = len(dir_list)
+
+            # print 'PREV_DIR: ' + str(prev_dir_list)
+
+            # print 'CURR_DIR: ' + str(dir_list)
+
+            # Current File is in Directory: Compare the given dir with previous Dir
+            if cur_dir and prev_dir != cur_dir:
+                # evaluate all matched items (from current and previous list)
+                matched = []
+                for element in dir_list:
+                    if element in prev_dir_list:
+                        matched.append(element)
+
+                # calculate difference between matched and length
+                diff = dir_depth - len(matched)
+
+                # print 'MATCHED:  ' + str(matched)
+
+                # if previous dir was not root
+                if prev_dir != '':
+                    # if the elements count is not equal we calculate the difference
+                    if len(dir_list) != len(prev_dir_list):
+                        dir_depth_prev = len(prev_dir_list)
+                        delta = dir_depth_prev - len(matched)
+
+                        for i in range(dir_depth_prev - delta, dir_depth_prev):
+                            output.append('</logicalFolder>')
+
+                    # if the elements count is equal, we subtract the matched length from the total length
+                    else:
+                        for i in range(len(matched), len(dir_list)):
+                            output.append('</logicalFolder>')
+
+                for i in range(dir_depth - diff, dir_depth):
+                    output.append('<logicalFolder name="f' + str(folder_count) + '" displayName="' + str(
+                        dir_list[i]) + '" projectFiles="true">')
+                    folder_count += 1
+
+            # Current File is in root
+            else:
+                # Close Tag if we are in root and the previous dir wasn't
+                if cur_dir == '' and prev_dir != '':
+                    for i in range(0, len(prev_dir_list)):
+                        output.append('</logicalFolder>')
+
+            # Save the Current Dir
+            prev_dir = cur_dir
+            output.append('<itemPath>' + str(item) + '</itemPath>')
+            # if last iteration close all open tags
+            if idx == len(file_list) - 1 and cur_dir != '':
```

Comment:
I have a refactor proposal:
Remove the `enumerate` from the loop, move `dir_depth` out of the loop and close all open tags after the loop. I think that may make for clearer code. For example:
```python
for item in file_list:
    ...
last_dir_len = len(os.path.dirname(file_list[-1]).split(os.path.sep))
output.append('</logicalFolder>' * last_dir_len)
return output
```

Reply:
Changed!